WooCommerce/Classes/ViewRelated/Products/ProductVariationDetailsFactory.swift

```diff
@@ -0,0 +1,59 @@
+import UIKit
+import Yosemite
+
+struct ProductVariationDetailsFactory {
+    /// Creates a view controller asynchronously that shows product variation details based on feature flags.
+    /// - Parameters:
+    ///   - productVariation: product variation model.
+    ///   - parentProduct: the parent product of the product variation.
+    ///   - presentationStyle: how the product variation details are presented.
+    ///   - currencySettings: site currency settings.
+    ///   - forceReadOnly: force the product variation details to be presented in read only mode.
+    ///   - onCompletion: called when the view controller is created and ready for display.
+    static func productVariationDetails(productVariation: ProductVariation,
+                                        parentProduct: Product,
+                                        presentationStyle: ProductFormPresentationStyle,
+                                        currencySettings: CurrencySettings = ServiceLocator.currencySettings,
+                                        forceReadOnly: Bool,
+                                        onCompletion: @escaping (UIViewController) -> Void) {
+        let vc = productVariationDetails(productVariation: productVariation,
+                                         parentProduct: parentProduct,
+                                         presentationStyle: presentationStyle,
+                                         currencySettings: currencySettings,
+                                         isEditProductsEnabled: forceReadOnly ? false: true,
+                                         isEditProductsRelease3Enabled: true,
```

Comment:
Do you think it's worth to pass `ServiceLocator.featureFlagService.isFeatureFlagEnabled(.editProductsRelease3)` here?
Just because we can remember easily when we will drop the `editProductsRelease3` feature flag to remove it.

Reply:
good point! updated in https://github.com/woocommerce/woocommerce-ios/pull/2940/commits/159456efb8123558650cd18a3e0719a5e27defd3